src/dfx/src/commands/deploy.rs

```diff
@@ -150,26 +161,44 @@ fn display_urls(env: &dyn Environment) -> DfxResult {
 
     if let Some(canisters) = &config.get_config().canisters {
         for (canister_name, canister_config) in canisters {
-            if config
+            let canister_is_remote = config
                 .get_config()
-                .is_remote_canister(canister_name, &network.name)?
-            {
+                .is_remote_canister(canister_name, &network.name)
+                .with_context(|| {
+                    format!(
+                        "Failed to determine if canister {} is remote.",
+                        canister_name
+                    )
+                })?;
+            if canister_is_remote {
                 continue;
             }
             let canister_id = match Principal::from_text(canister_name) {
                 Ok(principal) => Some(principal),
                 Err(_) => canister_id_store.find(canister_name),
             };
             if let Some(canister_id) = canister_id {
-                let canister_info = CanisterInfo::load(&config, canister_name, Some(canister_id))?;
+                let canister_info = CanisterInfo::load(&config, canister_name, Some(canister_id))
+                    .with_context(|| {
+                    format!(
+                        "Failed to load canister info for canister {}.",
+                        canister_name
+                    )
+                })?;
```

Comment:
Again a place where we aren't adding context; rather reporting information that the callee could report.